src/explorer/Components/ExplorationInfo.cs

```diff
@@ -0,0 +1,35 @@
+namespace Explorer.Components
+{
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+
+    using Explorer.Common;
+    using Explorer.Metrics;
+
+    public class ExplorationInfo
+        : ExplorerComponent<bool>, PublisherComponent
```

Comment:
I think we should avoid using primitive types as the type argument for `ExplorerComponent<>`. The type arg should be something unique to the component. 

For example, in this case there is another `ExplorerComponent<bool>` elsewhere in the code (EmailCheckComponent) - this will cause confusion when the dependency injector resolves class instances. (I suspect you might get strange results when you run this on a `text` column).

I would suggest defining a private `Result` or `ExplorationMetadata` class containing the Exploration info. We will add more items to this (I have a branch ready where we can insert the git commit hash and ref from the docker image).

Reply:
In order to avoid this problem in the future, would it be useful to add constraint on the parameter of the generic type `ExplorerComponent<TResult>`?

Reply:
Sorry, didn't see your comment, already approved and merged the PR... I think you're right, it might be useful to add the `class` constraint the type parameter.